cpu/cc2538/periph/spi.c

```diff
@@ -69,33 +73,16 @@ void spi_init(spi_t bus)
 
 void spi_init_pins(spi_t bus)
 {
-    switch ((uintptr_t)spi_config[bus].dev) {
-        case (uintptr_t)SSI0:
-            IOC_PXX_SEL[spi_config[bus].mosi_pin] = SSI0_TXD;
-            IOC_PXX_SEL[spi_config[bus].sck_pin ] = SSI0_CLK_OUT;
-            IOC_PXX_SEL[spi_config[bus].cs_pin  ] = SSI0_FSS_OUT;
-
-            IOC_SSIRXD_SSI0 = spi_config[bus].miso_pin;
-            break;
-
-        case (uintptr_t)SSI1:
-            IOC_PXX_SEL[spi_config[bus].mosi_pin] = SSI1_TXD;
-            IOC_PXX_SEL[spi_config[bus].sck_pin ] = SSI1_CLK_OUT;
-            IOC_PXX_SEL[spi_config[bus].cs_pin  ] = SSI1_FSS_OUT;
-
-            IOC_SSIRXD_SSI1 = spi_config[bus].miso_pin;
-            break;
-    }
-
-    IOC_PXX_OVER[spi_config[bus].mosi_pin] = IOC_OVERRIDE_OE;
-    IOC_PXX_OVER[spi_config[bus].miso_pin] = IOC_OVERRIDE_DIS;
-    IOC_PXX_OVER[spi_config[bus].sck_pin ] = IOC_OVERRIDE_OE;
-    IOC_PXX_OVER[spi_config[bus].cs_pin  ] = IOC_OVERRIDE_OE;
-
-    gpio_hardware_control(spi_config[bus].mosi_pin);
-    gpio_hardware_control(spi_config[bus].miso_pin);
-    gpio_hardware_control(spi_config[bus].sck_pin);
-    gpio_hardware_control(spi_config[bus].cs_pin);
```

Comment:
`gpio_hardware_control` isn't used anywhere else. How about we remove its definition? Or do you plan to cleanup cc2538/gpio anyway?